ts/src/heap-profiler-inspector.ts

```diff
@@ -0,0 +1,180 @@
+import * as inspector from 'node:inspector';
+import {AllocationProfileNode, Allocation} from './v8-types';
+
+const session = new inspector.Session();
+
+export interface SamplingHeapProfileSample {
+  size: number;
+  nodeId: number;
+  ordinal: number;
+}
+
+export interface SamplingHeapProfileNode {
+  callFrame: inspector.Runtime.CallFrame;
+  selfSize: number;
+  id: number;
+  children: SamplingHeapProfileNode[];
+}
+
+/**
+ * Need to create this interface since the type definitions file for node inspector
+ * at https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/inspector.d.ts
+ * has not been updated with the latest changes yet.
+ *
+ * The types defined through this interface are in sync with the documentation found at -
+ * https://chromedevtools.github.io/devtools-protocol/v8/HeapProfiler/
+ */
+export interface CompatibleSamplingHeapProfile {
+  head: SamplingHeapProfileNode;
+  samples: SamplingHeapProfileSample[];
+}
+
+export function startSamplingHeapProfiler(
+  heapIntervalBytes: number,
+  stackDepth: number
+): Promise<void> {
+  session.connect();
+  return new Promise<void>((resolve, reject) => {
+    session.post(
+      'HeapProfiler.startSampling',
+      {heapIntervalBytes},
+      (err: Error | null): void => {
+        if (err !== null) {
+          console.error(`Error starting heap sampling: ${err}`);
```

Comment:
Nothing should be logged within this repo (cloud-profiler-nodejs will log errors occurring; logging anything directly here will either result in things being double logged, or prevent users from being able to control what is logged)